---
title: "Class 10: Introduction to RCSB Protein Data Bank (PDB)"
author: "Ayanna Kerr (PID:A17143404)"
format: pdf
---

```{r}
data <- read.csv("Data Export Summary1.csv")
head(data)
```

> Q1: What percentage of structures in the PDB are solved by X-Ray and Electron Microscopy? 

```{r}
(sum(data$X.ray + data$EM)/sum(data$Total))*100
```


> Q2: What proportion of structures in the PDB are protein?

```{r}
sum(data$Total[1:3])/sum(data$Total)*100
```


> Q3: Type HIV in the PDB website search box on the home page and determine how many HIV-1 protease structures are in the current PDB?

2233

```{r}
library(bio3d)
read.pdb("1hsg.pdb")
```

> Q4: Water molecules normally have 3 atoms. Why do we see just one atom per water molecule in this structure?
Oxygen is the most electron-rich and most relevant in most interactions with other molecules. To avoid clutter, it makes sense to represent the most important atom in the water molecule, which is oxygen. 

> Q5: There is a critical “conserved” water molecule in the binding site. Can you identify this water molecule? What residue number does this water molecule have
H2O 308 

> Q6: Generate and save a figure clearly showing the two distinct chains of HIV-protease along with the ligand. You might also consider showing the catalytic residues ASP 25 in each chain and the critical water (we recommend “Ball & Stick” for these side-chains). Add this figure to your Quarto document.

```{r}
library(magick)

image_read("1HSG (4).png")

```


## Reading PDB file data into R

```{r}
pdb <- read.pdb("1hsg")
```

```{r}
pdb
```


> Q7: How many amino acid residues are there in this pdb object? 

198

> Q8: Name one of the two non-protein residues? 

HOH (127)

> Q9: How many protein chains are in this structure? 

2

```{r}
attributes(pdb)
```

```{r}
head(pdb$atom)
```

## Predicting functional motions of a single structure

```{r}
adk <- read.pdb("6s36")
```

```{r}
adk
```


```{r}
m <- nma(adk)
```

```{r}
plot(m)
```

```{r}
mktrj(m, file="adk_m7.pdb")
```

## Comparative structure analysis of adenylate kinase

Here we will work through a complete pipeline of analysis for a given protein family starting from one single database ID for a member of this family all the way to a deeper understanding of structure dynamics and function. 

```{r}
library(bio3d)
```

> Q10. Which of the packages above is found only on BioConductor and not CRAN?:
msa

> Q11. Which of the above packages is not found on BioConductor or CRAN?: 
Grantlab/bio3d-view

> Q12. True or False? Functions from the devtools package can be used to install packages from GitHub and BitBucket?:
TRUE

```{r}
library(bio3d)
id <- "1ake_A"
aa <- get.seq(id)
aa
```

> Q13. How many amino acids are in this sequence, i.e. how long is this sequence?:
214


```{r}
hits <- NULL
hits$pdb.id <- c('1AKE_A','6S36_A','6RZE_A','3HPR_A','1E4V_A','5EJE_A','1E4Y_A','3X2S_A','6HAP_A','6HAM_A','4K46_A','3GMT_A','4PZL_A')
```


```{r}
# Download related PDB files
files <- get.pdb(hits$pdb.id, path="pdbs", split=TRUE, gzip=TRUE)
```

```{r}
# Align related PDB files
pdbs <- pdbaln(files, fit = TRUE, exefile="msa")
```

```{r}
pdbs
```


```{r}
# Vector containing PDB codes for figure axis
ids <- basename.pdb(pdbs$id)

# Draw schematic alignment
#plot(pdbs, labels=ids)
```


## Annotate collected PDB stuctures

```{r}
anno <- pdb.annotate(ids)
unique(anno$source)
```

```{r}
anno
```

## Principle component analysis

```{r}
# Perform PCA
pc.xray <- pca(pdbs)
plot(pc.xray)
```

```{r}
# Calculate RMSD
rd <- rmsd(pdbs)

# Structure-based clustering
hc.rd <- hclust(dist(rd))
grps.rd <- cutree(hc.rd, k=3)

plot(pc.xray, 1:2, col="grey50", bg=grps.rd, pch=21, cex=1)
```

## Optional further visualization

```{r}
# Visualize first principal component
pc1 <- mktrj(pc.xray, pc=1, file="pc_1.pdb")
```

```{r}
#Plotting results with ggplot2
library(ggplot2)
library(ggrepel)

df <- data.frame(PC1=pc.xray$z[,1], 
                 PC2=pc.xray$z[,2], 
                 col=as.factor(grps.rd),
                 ids=ids)

p <- ggplot(df) + 
  aes(PC1, PC2, col=col, label=ids) +
  geom_point(size=2) +
  geom_text_repel(max.overlaps = 20) +
  theme(legend.position = "none")
p
```

## Normal Mode Analysis

```{r}
# NMA of all structures
modes <- nma(pdbs)
```

```{r}
plot(modes, pdbs, col=grps.rd)
```

> Q14. What do you note about this plot? Are the black and colored lines similar or different? Where do you think they differ most and why?:
There are two major distinct conformational states for Adk. They differ by collective low frequency displacement of two nucleotide-binding site regions that display distinct flexibilities upon nucleotide binding. The different colors represent different structures.  